### PR TITLE
Add XRRay and update XRInputPose and XRInputSource attribute names

### PR DESCRIPTION
**Pull request by jsantell**
Still need to actually test

### CODE REVIEW

**Comment by jsantell** [comment]
Fixed #14

**Comment by toji** [comment]
The change looks good in all technical respects, but I think we want to approach this a little more carefully. For one, XRRay hasn't actually been merged into the explainer yet, so I'd like to hold off on that portion of the change until we get group consensus. (Hopefully that'll come this week?)

Secondly, the input changes are necessary but if we land these changes now then the current samples will need to be updated and in doing so would break with Chrome 67/68's implementation. Those are known to be outdated at this point, so I'm happy to move past it, but ideally we'd want to provide some temporary shim layer that maps from the new behavior to the old while we still have those versions in the wild.

There's two logical places for that, and I'd love your opinion on it: We can bake that temporary shim layer into the polyfill so that anyone who uses the polyfill automatically gets the fixup, which seems good unless there's someone who wants to avoid the overhead of including the full polyfill? Alternatively, we could keep the polyfill focused on the latest spec state and I could implement a separate shim library that would probably live in the samples repo. Thoughts?

**Comment by toji** [comment]
FYI: The XRRay concept is now officially part of the explainer/spec, and the samples have been updated to use it. I'm in the process of rebasing this and making a few related minor updates, and then hopefully we can merge it.

**Comment by jsantell** [comment]
@toji sounds great! let me know when is good to orchestrate a new build/release with these changes.

**Comment by toji** [comment]
Okay, this has now been updated to reflect the current state of the spec and should be ready to merge whenever you get a chance to review.

One thing to note is the addition of a partial DOMPointReadOnly polyfill, which feels slightly out of place here and which I'm not exposing publicly for that reason. This is only there to ensure the polyfill continues to run on Edge and Safari, so I'm not inclined to make it into a full-fledged Geometry Interfaces polyfill.

File: src/api/XRInputPose.js [hunk 1]
```diff
@@ -13,33 +13,39 @@
...
    */
-  get emulatedPosition() { return this[PRIVATE].inputSourceImpl.emulatedPosition; }
+  get targetRay() { return this[PRIVATE].targetRay; }
 
   /**
-   * @return {Float32Array}
+   * TODO: Non-standard. Make this read-only in the future.
+   * @param {XRRay} value
+   */
+  set targetRay(value) { this[PRIVATE].targetRay = value; }
```
Comment (jsantell) [inline src/api/XRInputPose.js]: In the future, we can use the exposed `PRIVATE` symbols from #29 to set the targetRay in the GamepadXRInputSource component -- there's a few other spots where the public, exposed API differs from spec, and that change will allow us some greater access internally for these sorts of things -- mostly a comment to myself :smile:

File: src/lib/DOMPointReadOnly.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GLOBAL from './global';
+
+let domPointROExport = ('DOMPointReadOnly' in GLOBAL) ? DOMPointReadOnly : null;
+
+if (!domPointROExport) {
+  const PRIVATE = Symbol('@@webxr-polyfill/DOMPointReadOnly');
```
Comment (jsantell) [inline src/lib/DOMPointReadOnly.js]: We should non-conditionally export this symbol: `export const PRIVATE = Symbol('...')`

File: test/api/test-xr-ray.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+import mocha from 'mocha';
+import { assert } from 'chai';
+
+import XRRay from '../../src/api/XRRay.js';
+import DOMPointReadOnly from '../../src/lib/DOMPointReadOnly.js';
+
+describe('API - XRRay', () => {
+  it('defaults to <0,0,0> origin, <0,0,-1> direction', () => {
```
Comment (jsantell) [inline test/api/test-xr-ray.js]: Looking through the spec, I don't see any evidence that the direction should be <0,0,-1,0> by default (rather, I'd assume <0,0,0,0>) -- what should be the case here?

File: src/api/XRRay.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import DOMPointReadOnly from '../lib/DOMPointReadOnly';
+
+export default class XRRay {
+  /**
+   * @param {?XRVector3} origin
+   * @param {?XRVector3} direction
+   * @param {Float32Array} transformMatrix
+   */
+  constructor(origin=new DOMPointReadOnly(0, 0, 0, 1),
+              direction=new DOMPointReadOnly(0, 0, -1, 0),
```
Comment (jsantell) [inline src/api/XRRay.js]: Should direction's `z` be -1?

**Comment by toji** [comment]
There's nothing in the spec to say that should be a default, but given that it's required to be a normalized directional vector it should be _something_ with a length of 1. Having a vector that points directly into the screen (OpenGL convention being -Z) is probably the most logical choice in that case. Likely not a big issue either way since we really should never report an `XRInputPose` with a default `targetRay`.

**Pull request merged**